### ISSUE TITLE
Gradient Backward propogation

### ISSUE THREAD

**Issue by Mutual-Luo** [mayanbiao1234/Geometric-metrics-for-perceptual-manifolds#1]
Thank you for your great work.

but I have a problem: all your code uses numpy operation, how can you achieve the gradient backward propagation mentioned in your paper? For example, you use the curvature regularization.

**Comment by mayanbiao1234** [comment]
Hello, it is possible to convert it to pytorch version, for promotion I developed this toolkit using the most common numpy.&nbsp;This toolkit is mainly used for exploring deep neural networks, if you want to use it for training, please just convert it with pytorch's code statements. I haven't done any further organizing, I've been too busy lately.
&nbsp;


马彦彪
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月1日(星期天) 中午1:56
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; 
主题: [mayanbiao1234/Geometric-metrics-for-perceptual-manifolds] Gradient Backward propogation (Issue #1)















感谢你的伟大工作。

但我有一个问题：你所有的代码都使用numpy操作，你如何实现论文中提到的梯度反向传播？例如，你使用曲率正则化。

-
直接回复这封邮件，在GitHub上查看，或取消订阅.
***@***.*** &gt;

**Comment by Mutual-Luo** [comment]
Thank you very much!
